src/runtime/recipe/handle.ts

```diff
@@ -46,7 +47,26 @@ export class Handle implements Comparable<Handle> {
     this._recipe = recipe;
   }
 
-  _copyInto(recipe: Recipe, cloneMap: CloneMap, variableMap: VariableMap) {
+  toSlot(): Slot {
+    if (!this.type) {
+      return undefined;
+    }
+    const slotType = this.type.slandleType();
+    if (!slotType) {
+      return undefined;
+    }
+    const slotInfo = slotType.getSlot();
+
+    const slandle = new Slot(this.recipe, this.localName);
+    slandle.tags = this.tags;
+    slandle.id = this.id;
+    slandle.formFactor = slotInfo.formFactor;
+    // TODO cannot assign slandle handles as the slots do not actually track their handles but use a source particle connection mapping
+    // slandle.handles = [slotInfo.handle];
+    return slandle;
+  }
+
+  _copyInto(recipe: Recipe, _cloneMap: CloneMap, variableMap: VariableMap) {
```

Comment:
why add `_` prefix?

Reply:
_copyInfo needs the prefix to match the rest of the code base, I would like to remove this at some point.

The _ for the arguments has been discussed elsewhere (tslint complains) but I will stop listening to it about this.